plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java

```diff
@@ -138,96 +142,83 @@ private static IFile materialize(IProject project, AppEngineProjectConfig config
+    createChildFile("web.xml", Templates.WEB_XML_TEMPLATE, webInf, //$NON-NLS-1$
+        properties, monitor);
+  }
+
+  private static void createWebContents(IProject project, IProgressMonitor monitor)
+      throws CoreException {
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 10);
+
+    IFolder webapp = project.getFolder("src/main/webapp"); //$NON-NLS-1$
     createChildFile("index.html", Templates.INDEX_HTML_TEMPLATE, webapp, //$NON-NLS-1$
         Collections.<String, String>emptyMap(), subMonitor.newChild(5));
 
     copyChildFile("favicon.ico", webapp, subMonitor.newChild(5)); //$NON-NLS-1$
-
-    if (config.getUseMaven()) {
-      Map<String, String> mavenCoordinates = new HashMap<>();
-      mavenCoordinates.put("projectGroupId", config.getMavenGroupId()); //$NON-NLS-1$
-      mavenCoordinates.put("projectArtifactId", config.getMavenArtifactId()); //$NON-NLS-1$
-      mavenCoordinates.put("projectVersion", config.getMavenVersion()); //$NON-NLS-1$
-      if (isStandardProject) {
-        if (Objects.equal(AppEngineRuntime.STANDARD_JAVA_7.getId(), config.getRuntimeId())) {
-          mavenCoordinates.put("servletVersion", "2.5"); //$NON-NLS-1$ //$NON-NLS-2$
-          mavenCoordinates.put("compilerVersion", "1.7"); //$NON-NLS-1$ //$NON-NLS-2$
-        } else {
-          mavenCoordinates.put("servletVersion", "3.1"); //$NON-NLS-1$ //$NON-NLS-2$
-          mavenCoordinates.put("compilerVersion", "1.8"); //$NON-NLS-1$ //$NON-NLS-2$
-        }
-        createChildFile("pom.xml", Templates.POM_XML_STANDARD_TEMPLATE, //$NON-NLS-1$
-            project, mavenCoordinates, subMonitor.newChild(5));
-      } else {
-        createChildFile("pom.xml", Templates.POM_XML_FLEX_TEMPLATE, //$NON-NLS-1$
-            project, mavenCoordinates, subMonitor.newChild(5));
-      }
-    }
-
-    return hello;
   }
 
-  private static IFolder createFoldersForPackage(IFolder parentFolder,
-                                                 String packageName,
-                                                 SubMonitor subMonitor) throws CoreException {
-    IFolder folder = parentFolder;
-    if (packageName != null && !packageName.isEmpty()) {
-      String[] packages = packageName.split("\\.");  //$NON-NLS-1$
-      subMonitor.setWorkRemaining(packages.length);
-      for (int i = 0; i < packages.length; i++) {
-        folder = createChildFolder(packages[i], folder, subMonitor.newChild(1));
+  private static void createPomXml(IProject project, AppEngineProjectConfig config,
+      boolean isStandardProject, IProgressMonitor monitor) throws CoreException {
+    Map<String, String> properties = new HashMap<>();
+    properties.put("projectGroupId", config.getMavenGroupId()); //$NON-NLS-1$
+    properties.put("projectArtifactId", config.getMavenArtifactId()); //$NON-NLS-1$
+    properties.put("projectVersion", config.getMavenVersion()); //$NON-NLS-1$
+
+    String template;
```

Comment:
I found this a little hard to read. The template variable should at least go inside the outer if block to make the scope clearer.

Reply:
Done.